src/DiscoveryServerManager.cpp

```diff
@@ -713,6 +746,29 @@ void DiscoveryServerManager::loadProfiles(
     }
 }
 
+bool DiscoveryServerManager::loadProperties(tinyxml2::XMLElement* props_n)
+{
+    bool ret = true;
+    tinyxml2::XMLElement* prop = props_n->FirstChildElement(eprosima::fastrtps::DSxmlparser::PROPERTY);
+
+    for (;prop != nullptr; prop = prop->NextSiblingElement(eprosima::fastrtps::DSxmlparser::PROPERTY))
+    {
+        tinyxml2::XMLElement* name = prop->FirstChildElement(eprosima::fastrtps::DSxmlparser::NAME);
+        tinyxml2::XMLElement* value = prop->FirstChildElement(eprosima::fastrtps::DSxmlparser::VALUE);
+
+        if (nullptr != name && nullptr != value)
+        {
+            properties_.push_back({name->GetText(), value->GetText()});
+        }
+        else
+        {
+            ret = false;
```

Comment:
Maybe we could be a little more specific on what the failure is with a LOG_ERROR (missing name/value)